picoquic/logger.c

```diff
@@ -1821,7 +1821,7 @@ void picoquic_open_cc_dump(picoquic_cnx_t * cnx)
         const size_t suffix_len = strlen(suffix);
         size_t folder_length = strlen(cnx->quic->cc_log_dir);
 
-        if (folder_length + 1 + 2*cnx->initial_cnxid.id_len + suffix_len + 1 > sizeof(cc_log_file_name)) {
+        if (folder_length + (size_t)1 + ((size_t)2)* (size_t)cnx->initial_cnxid.id_len + suffix_len + (size_t)1 > sizeof(cc_log_file_name)) {
```

Comment:
Arithmetic operators in the C language are left-to-right associative, the first argument determines the type the operation is performed on. That being said, (size_t)1 shouldn't be necessary and it should be sufficient to have:
```
if (folder_length + 1 + (size_t)cnx->initial_cnxid.id_len * 2 + suffix_len + 1 > sizeof(cc_log_file_name)) {
```

Reply:
Yes, I probably went too far there.